aiida/orm/implementation/entities.py

```diff
@@ -0,0 +1,354 @@
+# -*- coding: utf-8 -*-
```

Comment:
I think I like this new file, but maybe it makes sense to also move the `BackendEntity` class itself here, from the `backends` module?

Reply:
How about `BackendCollection`? Does it fit better in the `entities` module as well, or should it remain in the `backends` one?

Reply:
Given that it is a collection of entities and not backends, I think you make a fair point. But I see that you already moved it :+1: